### ISSUE TITLE
Pruing/Turning Filters off

### ISSUE THREAD

**Issue by megazostrodon** [keras-team/keras#10814]
Hello everyone,

I am looking for an easy way to prune away filters and/or to turn them off (e.g. set their values to zero).

So far, I have trained a model, and then retrieved the weights by 

weights = model.get_weights()

subsequently, I set a filter's values to zero, by 

weights[n][0] = -0.0 # filter, i.e. an even number
weights[n+1] = -0.0 # corresponding bias

and initialize a new model with those weights by

newmodel.set_weights(weights)

However, after training, they are not zero, so the weights get updated, although there should not be any gradient. I have been using relu as an activation function and SGD (with no momentum) as optimizer.

Can somebody suggest an (intuitive) way how to achieve this or give an explanation why it does not work this way?

Thank you in advance,
megazostrodon

**Comment by gabrieldemarmiesse** [comment]
The weights get updated. It's the purpose of a neural networks. If you want to do it, I would advise to write again a convolutional layer. Create a second kernel, same size as the first, but make this kernel non trainable. The second kernel should only have ones and zeros (you set them manually). Then, before doing the convolution step, multiply (elementwise) the two kernels together, resulting in the behaviour that you wanted.
